### PR TITLE
Save chat for a target.

### PR DESCRIPTION
**Pull request by SinghHimadri**
## Issue that this pull request solves
#343
Closes: # (issue number)

## Proposed changes

Brief description of what is fixed or changed
As the export chat option is not available on Whatsapp web and also couldn't find a good reference from the internet for this issue I came up with this idea.Run using: wplay -wsc <target>. currently, it doesn't save whole chat but up to some extend. I have used the selectors to reach message sender and time and modified it accordingly and saving the chats as a .txt file in wplay/saveChats folder. I couldn't think of any other way.
## Types of changes

_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface
Screenshot from 2020-05-23 20-45-15

Screenshot from 2020-05-23 20-45-51

## Other information

Any other information that is important to this pull request

### CODE REVIEW

File: wplay/__main__.py [hunk 1]
```diff
@@ -174,15 +174,17 @@ async def get_and_match_args(parser):
...
-        except (IndexError, ValueError):
-            bID: int = 0
-        save_chat.runMain('pull', str(args.target), bID)
+             parser.print_help()
+             parser.exit()
+         try:
+             bID: int = int(sys.argv[3])
+         except (IndexError, ValueError):
+             bID: int = 0
+         save_chat.runMain('pull', str(args.target), bID)'''
```
Comment (rpotter12) [inline wplay/__main__.py:187]: Codacy Issue found: [String statement has no effect](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5574725)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity decreasing per file
==============================
+ wplay/save_chat.py  -8
+ wplay/__main__.py  -2
         

Clones added
============
- wplay/save_chat.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5574725&bid=18276334)

File: wplay/save_chat.py [hunk 2]
```diff
@@ -190,4 +190,61 @@ def runMain(mode, asset, bID):
             if len(drives) > 1:
                 print('Backup: ' + str(i))
                 folder = 'WhatsApp-' + str(i)
-            getMultipleFiles(drive[1], folder)
+            getMultipleFiles(drive[1], folder)'''
```
Comment (rpotter12) [inline wplay/save_chat.py:193]: Codacy Issue found: [String statement has no effect](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5574725)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity decreasing per file
==============================
+ wplay/save_chat.py  -8
+ wplay/__main__.py  -2
         

Clones added
============
- wplay/save_chat.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5574725&bid=18276334)

File: wplay/save_chat.py (hunk 2 above)
Comment (rpotter12) [inline wplay/save_chat.py:193]: Codacy Issue found: [String statement has no effect](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5574725)

File: wplay/__main__.py [hunk 3]
```diff
@@ -174,15 +174,19 @@ async def get_and_match_args(parser):
...
-            bID: int = 0
-        save_chat.runMain('pull', str(args.target), bID)
+             parser.print_help()
+             parser.exit()
+         try:
+             bID: int = int(sys.argv[3])
+         except (IndexError, ValueError):
+             bID: int = 0
+         save_chat.runMain('pull', str(args.target), bID)
+        '''
```
Comment (rpotter12) [inline wplay/__main__.py:189]: Codacy Issue found: [String statement has no effect](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5574725)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity decreasing per file
==============================
+ wplay/save_chat.py  -8
+ wplay/__main__.py  -2
         

Clones added
============
- wplay/save_chat.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5574725&bid=18276334)

File: wplay/save_chat.py [hunk 4]
```diff
@@ -191,3 +191,61 @@ def runMain(mode, asset, bID):
                 print('Backup: ' + str(i))
                 folder = 'WhatsApp-' + str(i)
             getMultipleFiles(drive[1], folder)
+'''
```
Comment (rpotter12) [inline wplay/save_chat.py:194]: Codacy Issue found: [String statement has no effect](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5574725)

File: wplay/__main__.py (hunk 3 above)
Comment (rpotter12) [inline wplay/__main__.py:189]: Codacy Issue found: [String statement has no effect](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5574725)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity decreasing per file
==============================
+ wplay/__main__.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5574725&bid=18276334)

File: wplay/__main__.py [hunk 5]
```diff
@@ -174,15 +174,19 @@ async def get_and_match_args(parser):
...
-            parser.exit()
-        try:
-            bID: int = int(sys.argv[3])
-        except (IndexError, ValueError):
-            bID: int = 0
-        save_chat.runMain('pull', str(args.target), bID)
+    elif args.save_chat:
+        await save_chat.save_chat(args.target)
+
+
```
Comment (rpotter12) [inline wplay/__main__.py:180]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5574725)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity decreasing per file
==============================
+ wplay/__main__.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5574725&bid=18276334)

File: wplay/__main__.py [hunk 6]
```diff
@@ -174,15 +174,19 @@ async def get_and_match_args(parser):
...
+        # if args.target is None:
+        #      parser.print_help()
+        #      parser.exit()
+        #  try:
+        #      bID: int = int(sys.argv[3])
+        #  except (IndexError, ValueError):
+        #      bID: int = 0
+        #  save_chat.runMain('pull', str(args.target), bID)
+
```
Comment (rpotter12) [inline wplay/__main__.py:189]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5574725)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ wplay/__main__.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5574725&bid=18276334)

**Comment by SinghHimadri** [comment]
@rpotter12 Please review my PR.

**Pull request merged**